### PR TITLE
refactor(ivy): correct circular dependency which causes g3 failures

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

**Review by kara — APPROVED** [review]
LGTM

File: packages/core/src/debug/debug_node.ts [hunk 1]
```diff
@@ -11,16 +11,18 @@ import {getViewComponent} from '../render3/global_utils_api';
 import {CONTAINER_HEADER_OFFSET, LContainer, NATIVE} from '../render3/interfaces/container';
 import {TElementNode, TNode, TNodeFlags, TNodeType} from '../render3/interfaces/node';
 import {StylingIndex} from '../render3/interfaces/styling';
+import {isComponent, isLContainer} from '../render3/interfaces/type_checks';
```
Comment (kara) [inline packages/core/src/debug/debug_node.ts]: Wait a sec, does this file exist? I think it's missing from the diff.

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/core/src/debug/debug_node.ts (hunk 1 above)
Comment (mhevery) [reply packages/core/src/debug/debug_node.ts]: sorry, fixed

**Review by kara — APPROVED** [review]
LGTM

File: packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts [hunk 2]
```diff
@@ -2330,7 +2330,7 @@ describe('compiler compliance', () => {
         }
         if (rf & 2) {
           const $item$ = $i0$.ɵɵnextContext().$implicit;
-          const $foo$ = $i0$.ɵɵreference(2);
+          const $foo$ = $i0$.ɵ����reference(2);
```
Comment (kara) [inline packages/compiler-cli/test/compliance/r3_compiler_compliance_spec.ts]: Revert? Causing CI issues

**Comment by kara** [comment]
[Ivy presubmit](https://test.corp.google.com/ui#id=OCL:259038795:BASE:259038948:1563571649459:4e423c30)

**Comment by kara** [comment]
merge-assistance: global approval

**Pull request closed without merging**